src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -696,7 +696,12 @@ private Point2D CalculatePortPosition(PortModel portModel, int proxyPortIndex)
             {
                 case PortType.Input:
                     return new Point2D(Left, y);
-                case PortType.Output:         
+                case PortType.Output:
+                    if (portModel.Owner is CodeBlockNodeModel)
+                    {
+                        // Special case because code block outputs are smaller than regular outputs.
+                        return new Point2D(Left + Width, y + 6);
```

Comment:
Assuming there is a shift in this case, is there a way we can get the port to be closer to the group edge?

Reply:
can this value be based on a ratio instead of an exact pixel value?

Reply:
@QilongTang If we adjust the margin more, it will displace ports coming from other nodes. This margin has been adjusted considering the area used for port snapping.